sklearn/neighbors/base.py

```diff
@@ -283,6 +283,15 @@ def _pairwise(self):
         return self.metric == 'precomputed'
 
 
+def _tree_query_parallel_helper(tree, data, n_neighbors, return_distance):
+    """Helper for the Parallel calls in KNeighborsMixin.kneighbors
+
+    The Cython method tree.query is not directly picklable by cloudpickle
+    under PyPy.
```

Comment:
Interesting that this works while pickling directly fails. I don't really have a good understanding of cloudpickle (or pickling in general) to have much to say about this. LGTM since it fixes the original failure.